File: hackerrank/easy/0032-the-birthday-bar/swift/solution.swift
// solution.swift

func birthday(s: [Int], d: Int, m: Int) -> Int {
    guard s.count >= m else { return 0 }
    var count = 0
    for i in 0...(s.count - m) {
        var sum = 0
        for j in 0..<m {
            sum += s[i + j]
        }
        if sum == d { count += 1 }
    }
    return count
}

// --- demo run (samples) ---
print(birthday(s: [1, 2, 1, 3, 2], d: 3, m: 2))  // 2
print(birthday(s: [1, 1, 1, 1, 1, 1], d: 3, m: 2))  // 0
print(birthday(s: [4], d: 4, m: 1))  // 1

File: hackerrank/easy/0032-the-birthday-bar/python/solution.py
## solution.py

from typing import List

def birthday(s: List[int], d: int, m: int) -> int:
    if len(s) < m:
        return 0
    count = 0
    for i in range(len(s) - m + 1):
        if sum(s[i:i+m]) == d:
            count += 1
    return count


# --- demo run (samples) ---
if __name__ == "__main__":
    print(birthday([1, 2, 1, 3, 2], 3, 2))   # 2
    print(birthday([1, 1, 1, 1, 1, 1], 3, 2)) # 0
    print(birthday([4], 4, 1))               # 1